tests/aws/services/s3/test_s3.py

```diff
@@ -340,14 +340,16 @@ def test_copy_object_kms(self, s3_bucket, kms_create_key, snapshot, aws_client):
     @markers.snapshot.skip_snapshot_verify(paths=["$..AccessPointAlias"])
     def test_region_header_exists(self, s3_create_bucket, snapshot, aws_client):
         snapshot.add_transformer(snapshot.transform.s3_api())
+        region = "eu-west-1"
+        snapshot.add_transformer(RegexTransformer(region, "<region>"))
         bucket_name = s3_create_bucket(
-            CreateBucketConfiguration={"LocationConstraint": "eu-west-1"},
+            CreateBucketConfiguration={"LocationConstraint": region},
         )
         response = aws_client.s3.head_bucket(Bucket=bucket_name)
-        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == "eu-west-1"
+        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == region
         snapshot.match("head_bucket", response)
         response = aws_client.s3.list_objects_v2(Bucket=bucket_name)
-        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == "eu-west-1"
+        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == region
         snapshot.match("list_objects_v2", response)
```

Comment:
I would replace the test with the following:
```python
    @markers.aws.validated
    def test_region_header_exists(self, s3_create_bucket_with_client, aws_client_factory):
        region_default = "us-east-1"
        region_non_default = "eu-west-1"
        client_default_region = aws_client_factory(region_name=region_default).s3
        client_non_default_region = aws_client_factory(region_name=region_non_default).s3

        bucket_name = f"test-bucket-{short_uid()}"
        s3_create_bucket_with_client(
            client_default_region,
            Bucket= bucket_name,
            CreateBucketConfiguration={"LocationConstraint": region_non_default},
        )
        response = client_default_region.head_bucket(Bucket= bucket_name)
        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == region_non_default

        response = client_default_region.list_objects_v2(Bucket= bucket_name)
        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == region_non_default

        response = client_non_default_region.head_bucket(Bucket= bucket_name)
        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == region_non_default

        response = client_non_default_region.list_objects_v2(Bucket= bucket_name)
        assert response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"] == region_non_default
```

It would not be impacted anymore by the multi account test changes that way.